Update project list view to readily show active notebooks

jeff-phillips-18 opened the pull request:
Closes [RHOAIENG-14674](https://issues.redhat.com/browse/RHOAIENG-14674)

## Description
PM review revealed the inability to quickly find active workbenches and the need for too many click to shutdown active workbenches. Updated the Project list view to show active workbenches in the workbenches column. Added start/stop actions to each work bench row rather than have them in the kebab menu.

## How Has This Been Tested?
Navigate to the Data Science Project page.
Verify the workbench counts for stopped or active workbenches are accurate (workbenches in the `starting` phase are counted as active).
Expand the workbenches column for projects with active and stopped workbenches.
Verify the correct action is shown (start or stop) based on the workbench status.

## Test Impact
Updated the e2e tests to check for the correct counts and actions

## Screen shots

![image](https://github.com/user-attachments/assets/592ff797-14bc-472d-b337-d9ff3936e1f7)


![image](https://github.com/user-attachments/assets/ba49985c-8adf-4cb8-9420-027838a61461)


![image](https://github.com/user-attachments/assets/111a069c-aa81-4e7b-883e-d9452713a678)

## Request review criteria:
Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

christianvogt commented:
Changes worked well except for the case where the workbench is started or stopped from the project list page. The state never seems to transition to `running` or `stopped`. It gets stuck in `starting` or `stopping` until the page is refreshed.

jeff-phillips-18 commented:
@christianvogt I've updated the page to watch the notebook states (updating every 30 seconds). Also fixed an issue where the stop modal was un-cancelable.

jgiardino commented:
Testing the latest updates, I see an issue on the Workbenches tab. When I click Stop for a Running or Starting workbench, I see a confirmation modal. But the status behind the modal switches to Stopping at the same time, without letting me cancel out of that action in the confirmation modal.

![image](https://github.com/user-attachments/assets/6b980ff9-03b2-4acf-b31c-58602e2c3ce9)

jeff-phillips-18 commented:
Thanks @jgiardino. Pushed the fix.

jgiardino commented:
I tested again. Overall these updates look good and I'm not able to find any issues. There is one odd behavior I noticed but cannot reproduce, where I opened a workbench that was in a Running state, but the JN didn't open, and then when I returned to ODH I saw that the Running state had changed to a Stopped state. 

The first time I saw this behavior, the following message displayed
![image](https://github.com/user-attachments/assets/f8cae27d-60a6-462f-8206-4d8b87d421ed)

The second time I saw this behavior, the following message displayed
![image](https://github.com/user-attachments/assets/c2ac8ac0-10bc-4d47-bab6-60fb55cee60f)

kywalker-rh reviewed (approved):
lgtm... looks and works as expected

christianvogt commented:
/lgtm
/approve
